src/systems/variational_smoother_constraint.C

```diff
@@ -521,6 +521,51 @@ void VariationalSmootherConstraint::constrain_node_to_line(const Node & node,
     }
 }
 
+void VariationalSmootherConstraint::find_nodal_or_face_neighbors(
+    const MeshBase &mesh, const Node &node,
+    const std::unordered_map<dof_id_type, std::vector<const Elem *>>
+        &nodes_to_elem_map,
+    std::vector<const Node *> &neighbors)
+{
+  // Find all the nodal neighbors... that is the nodes directly connected
+  // to this node through one edge.
+  MeshTools::find_nodal_neighbors(mesh, node, nodes_to_elem_map, neighbors);
+
+  // If no neighbors are found, use all nodes on the containing side as
+  // neighbors.
+  if (!neighbors.size())
+    {
+      // Grab the element containing node
+      // It doesn't make sense for a node having no edge neighbors to belong to
+      // multiple elements, right? Otherwise, we will figure out those cases as
+      // they occur.
```

Comment:
Won't any face-centered node on the "boundary" between two subdomains belong to two elements?

Reply:
But you've already instantiated the HEX27 tests!?  I'd have expected that to be the iconic example of the problem.

Reply:
Yeah, so these high order hex tests take a bit to run in dbg mode, so I always ran them in opt and didn't see the assertion hit in dbg until I put this PR up. I don't know what I was thinking when I wrote the assertion. I took it out.